src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs

```diff
@@ -103,12 +103,7 @@ internal object SyncRoot
                     throw new NotSupportedException(SR.NotSupported_DynamicAssembly);
                 }
 
-                string? codeBase = GetCodeBase();
-                if (codeBase is null)
-                {
-                    // Not supported if the assembly was loaded from single-file bundle.
-                    throw new NotSupportedException(SR.NotSupported_CodeBase);
-                }
+                string? codeBase = GetCodeBase() ?? throw new NotSupportedException(SR.NotSupported_CodeBase);
```

Comment:
```suggestion
                string codeBase = GetCodeBase() ?? throw new NotSupportedException(SR.NotSupported_CodeBase);
```

Reply:
I'd actually initially made that change before noticing that a few lines down the local is assigned to something that may be null. It's just outside of the diff GH shows.